scripts/adlsgen2setup.py

```diff
@@ -0,0 +1,166 @@
+import argparse
+import asyncio
+import os
+
+import aiohttp
+from azure.core.credentials_async import AsyncTokenCredential
+from azure.identity.aio import AzureDeveloperCliCredential
+from azure.storage.filedatalake.aio import (
+    DataLakeDirectoryClient,
+    DataLakeServiceClient,
+)
+
+
+class AdlsGen2Setup:
+    async def __aenter__(self):
+        self.service_client = DataLakeServiceClient(
+            account_url=f"https://{self.storage_account_name}.dfs.core.windows.net", credential=self.credentials
+        )
+        await self.service_client.__aenter__()
+        return self
+
+    async def __aexit__(self, *args, **kwargs):
+        await self.service_client.__aexit__(*args, **kwargs)
+
+    def __init__(
+        self,
+        data_directory: str,
+        storage_account_name: str,
+        filesystem_name: str,
+        security_enabled_groups: bool,
+        credentials: AsyncTokenCredential,
+        verbose: bool = False,
+    ):
+        self.data_directory = data_directory
+        self.storage_account_name = storage_account_name
+        self.filesystem_name = filesystem_name
+        self.credentials = credentials
+        self.security_enabled_groups = security_enabled_groups
+        self.verbose = verbose
+
+    async def run(self):
+        token_result = await self.credentials.get_token("https://graph.microsoft.com/.default")
+        self.graph_headers = {"Authorization": f"Bearer {token_result.token}"}
+        if self.verbose:
+            print("Ensuring gptkbcontainer exists...")
+        async with self.service_client.get_file_system_client("gptkbcontainer") as filesystem_client:
+            if not await filesystem_client.exists():
+                await filesystem_client.create_file_system()
+
+            if self.verbose:
+                print("Creating groups...")
+            admin_id = await self.create_or_get_group("GPTKB_AdminTest")
+            hr_id = await self.create_or_get_group("GPTKB_HRTest")
+            employee_id = await self.create_or_get_group("GPTKB_EmployeeTest")
+
+            if self.verbose:
+                print("Ensuring benefitinfo and employeeinfo directories exist...")
```

Comment:
Are these hardcoded for this example? How would someone customize?

Reply:
The problem with the access control lists are that they are specific to the user. What I can do is add a helper function that takes a list of files / their acls rather than hardcoding it this way, and take some info as parameters. The hardcoded behavior is useful for demonstrating a complete flow with the example, let me know if you disagree.

Reply:
Ah okay, could you make it so that the class/function accepts a data type (dict or list) and then for the demo, we pass in the data either via shell script args, acls.csv or acl.json file?